core/src/c_api/tiledb.cc

```diff
@@ -786,64 +731,60 @@ int tiledb_array_iterator_get_value(
     int attribute_id,
     const void** value,
     size_t* value_size) {
-  // Sanity check
-  if (!sanity_check(tiledb_array_it))
+  TileDB_CTX* ctx = tiledb_array_it->ctx_;
+
+  if (save_error(ctx, sanity_check(tiledb_array_it)))
     return TILEDB_ERR;
 
   // Get value
-  tiledb::Status st =
-      tiledb_array_it->array_it_->get_value(attribute_id, value, value_size);
-  if (!st.ok()) {
-    strcpy(tiledb_errmsg, st.to_string().c_str());
+  if (save_error(
+          ctx,
+          tiledb_array_it->array_it_->get_value(
+              attribute_id, value, value_size)))
     return TILEDB_ERR;
-  }
 
-  // Success
   return TILEDB_OK;
 }
 
 int tiledb_array_iterator_next(TileDB_ArrayIterator* tiledb_array_it) {
-  // Sanity check
-  if (!sanity_check(tiledb_array_it))
+  TileDB_CTX* ctx = tiledb_array_it->ctx_;
```

Comment:
What if `tiledb_array_it` in `nullptr`? Same for all or most functions in the C API (I will omit repeating this comment).